### PR TITLE
tests: agnhost image updates

### PR DESCRIPTION
**Pull request by claudiubelu**
**What type of PR is this?**

/kind failing-test
/kind feature

/sig testing
/sig windows

**What this PR does / why we need it**:

Adds the following subcommands to the agnhost binary:

- dns-server-list: outputs the host's DNS server list with which it was
  configured with. This can be found in /etc/resolv.conf on Linux and
  through some powershell commands on Windows.
- etc-hosts: outputs the host's hosts file. This can be found in /etc/hosts
  for Linux, and C:/Windows/System32/drivers/etc/hosts for Windows.
- pause: pauses the binary's execution. This can be used to keep the Pod in
  a Running state for various reasons, including executing additional agnhost
  commands.

Refactors bits of the code to avoid duplication.

Adds a README for the agnhost image.

**Which issue(s) this PR fixes**:

Related #73425
Related #74985
Needed by: 76508

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: test/images/agnhost/README.md [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+
+The image was created for testing purposes, reducing the need for having different test
+cases for the same tested behaviour.
+
+## Usage
+
+The ``agnhost`` binary is a CLI with the following subcommands:
+
+- ``dns-suffix``: It will output the host's configured DNS suffix list, separated by commas.
+- ``dns-server-list``: It will output the host's configured DNS servers, separated by comnmas.
```
Comment (neolit123) [inline test/images/agnhost/README.md]: type `comnmas`

File: test/images/agnhost/README.md [hunk 2]
```diff
@@ -0,0 +1,77 @@
+# Agnhost
+
+## Overview
+
+There are significant differences between Linux and Windows, especially in the way
+something can be obtained or tested. For example, the DNS suffix list can be found in
+``/etc/resolv.conf`` on Linux, but on Windows, such file does not exist, the same
```
Comment (neolit123) [inline test/images/agnhost/README.md]: i think the double quotes can be replaced with single ones in the new content.

File: test/images/agnhost/README.md [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+- ``pause``: It will pause the execution of the binary. This can be used for containers
+  which have to be kept in a ``Running`` state for various purposes, including executing
+  other ``agnhost`` commands.
+- ``help``: Prints the binary's help menu. Additionally, it can be followed by another
+  subcommand in order to get more information about that subcommand, including its
+  possible arguments.
+
+For example, let's consider the following ``pod.yaml`` file:
+
+    apiVersion: v1
```
Comment (neolit123) [inline test/images/agnhost/README.md]: the yaml can/should be quoted around triple quotes - e.g. ```yaml

**Comment by spiffxp** [comment]
/approve
I leave the `/lgtm` to @neolit123 

Ping #sig-testing once this has merged to ensure a new image is built using these changes

**Pull request merged**